coordinator/src/services/validation/bundle-validator.ts

```diff
@@ -21,121 +22,46 @@ export function verifyBundle(bundle: fhir.Bundle): Array<errors.ValidationError>
-        "https://fhir.nhs.uk/R4/StructureDefinition/Extension-DM-prescriptionType",
-        "MedicationRequest" +
-        '.extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-prescriptionType")'
-      )
-    },
-    {
-      fieldName: 'extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-DM-ResponsiblePractitioner")',
-      fieldAccessor: (medicationRequest) => getExtensionForUrlOrNull(
-        medicationRequest.extension,
-        "https://fhir.nhs.uk/R4/StructureDefinition/Extension-DM-ResponsiblePractitioner",
-        "MedicationRequest" +
-        '.extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-DM-ResponsiblePractitioner")'
-      )
-    },
-    {
-      fieldName: 'extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-UKCore-MedicationRepeatInformation")',
-      fieldAccessor: (medicationRequest) => getExtensionForUrlOrNull(
-        medicationRequest.extension,
-        "https://fhir.nhs.uk/R4/StructureDefinition/Extension-UKCore-MedicationRepeatInformation",
-        "MedicationRequest" +
-        '.extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-UKCore-MedicationRepeatInformation")'
-      )
-    }
+  const fhirPaths = [
+    "groupIdentifier",
+    "category",
+    "authoredOn",
+    "subject",
+    "requester",
+    "dispenseRequest.performer",
+    "dispenseRequest.validityPeriod",
+    "dispenseRequest.expectedSupplyDuration",
+    'dispenseRequest.extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-performerSiteType")',
+    'extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-prescriptionType")',
+    'extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-DM-ResponsiblePractitioner")',
+    'extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-UKCore-MedicationRepeatInformation")'
   ]
   const medicationRequests = getMedicationRequests(bundle)
-
-  return consistencyValidators.map(
-    (validationCondition) => verifyValueIdenticalForAllMedicationRequests(
-      medicationRequests,
-      validationCondition.fieldName,
-      validationCondition.fieldAccessor
-    )
-  ).filter(notEmpty)
-}
-
-function notEmpty<T>(value: T | null | undefined): value is T {
-  return value !== null && value !== undefined
-}
-
-function verifyValueIdenticalForAllMedicationRequests<U>(
-  medicationRequests: Array<fhir.MedicationRequest>,
-  fieldName: string,
-  fieldAccessor: (resource: fhir.MedicationRequest) => U
-): errors.ValidationError | null {
-  const fieldValues = medicationRequests.map(fieldAccessor)
-  const serializedFieldValues = fieldValues.map(value => LosslessJson.stringify(value))
-  const uniqueFieldValues = new Set(serializedFieldValues)
-  return uniqueFieldValues.size === 1 ? null : new errors.MedicationRequestValueError(fieldName, [...uniqueFieldValues])
+  return fhirPaths
+    .map((fhirPath) => verifyIdenticalForAllMedicationRequests(bundle, medicationRequests, fhirPath))
+    .filter(Boolean)
```

Comment:
Is this actually filtering out all non-null values?

Reply:
The tests pass so I'm going to say yes

Reply:
I think we use this in other places too